refactor-group/refactor-platform-fe
# Improve dashboard & coaching session page data fetching and caching

## Description
This PR makes a significant jump in terms of the maturity of the pattern of fetching and caching backend entity data on the frontend client side.

It now uses SWR for most entity fetching (still using Axios for the actual fetch).

It also updates to use React@19 and Next@15.

**Note:** it's a bit of a large PR, but it's really not that complex. The same pattern is repeated for each major entity type.

#### GitHub Issue: none

### Changes
* Replaces `AppStateStore` with `OrganizationStateStore`, `CoachingRelationshipStateStore`, `CoachingSessionStateStore`
* Uses SWR for a much cleaner data fetch layering with components, many fewer `useEffect` uses
* Updates to React@19
* Updates to Next@15

### Screenshots / Videos Showing UI Changes (if applicable)
N/A


### Testing Strategy
1. Login and run through every bit of functionality
2. Make sure everything behaves exactly the same as on `main` branch


### Concerns
None
